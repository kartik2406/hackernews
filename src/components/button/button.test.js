import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Button from "./button";

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {
  const props = {
    onClick: () => {}
  };
  it("should render without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button {...props}> More </Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("should have a valid snapshot", () => {
    const component = renderer.create(<Button {...props}> More </Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should have single button tag", () => {
    const element = render(<Button {...props}>More</Button>);
    expect(element.text()).toBe("More");
  });
});
