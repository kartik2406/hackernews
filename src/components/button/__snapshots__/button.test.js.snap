// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have a valid snapshot 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
   More 
</button>
`;
